src/main/scala/org/broadinstitute/dsde/workbench/sam/db/tables/ResourceRoleTable.scala

```diff
@@ -0,0 +1,21 @@
+package org.broadinstitute.dsde.workbench.sam.db.tables
+
+import org.broadinstitute.dsde.workbench.sam.db.{DatabaseKey, SamTypeBinders}
+import org.broadinstitute.dsde.workbench.sam.model.ResourceRoleName
+import scalikejdbc._
+
+final case class ResourceRolePK(value: Long) extends DatabaseKey
+final case class ResourceRoleRecord(id: ResourceRolePK,
+                                    resourceTypeId: ResourceTypePK,
+                                    role: ResourceRoleName)
```

Comment:
For unique constraints, we did something like this in Hamm: https://github.com/DataBiosphere/hamm/blob/master/core/src/main/scala/org/broadinstitute/dsp/workbench/hamm/db/JobTable.scala#L91

Not saying we need to do that here - but it _is_ difficult to know if there are any unique keys on a table just from looking at the class definitions the way it is. But there might be better ways of doing it and we could add that later.

Reply:
After some discussion, we're going to hold off on implementing unique keys like Hamm did for now. There is agreement (at least between me and Anu) that the record case classes don't do a good job of representing some of the underlying constraints that are described in the Liquibase changeset. The unique key case classes that Hamm uses would improve that representation, but they would not have any other purpose at this point in time which is why we won't be implementing them right now. That said, we might want to consider ways to improve how well the case classes represent the underlying tables.